src/main/java/net/minecraftforge/common/DimensionManager.java

```diff
@@ -199,14 +209,49 @@ public static ServerWorld getWorld(MinecraftServer server, DimensionType dim, bo
         return ret;
     }
 
+    /**
+     * Marks the given dimension for deletion upon next reload of the save.
+     * The corresponding data directory will be deleted as well.
+     *
+     * Note that the dimension need not be currently unloaded. The dimension will stay functioning as normal
+     * until the save is reloaded.
+     *
+     * Vanilla dimensions are not supported and will throw an exception.
+     *
+     * @param dim the dimension to delete
+     */
+    public static void markForDeletion(DimensionType dim)
+    {
+        if (dim.isVanilla())
+        {
+            throw new IllegalArgumentException("Cannot delete vanilla dimensions.");
+        }
+        getData(dim).markedForDeletion = true;
+
+        Path base = Paths.get(".").toAbsolutePath().normalize();
```

Comment:
This is now how you get the current directory, we have FMLPaths which should have the supplied --gamedir paths.
Nothing in Minecrat should rely on the current execution directory directly. {Server's should set the gamedir to CWD if none is supplied}

Reply:
It doesn't matter what the path is. This is simply to check that the dimension returns a proper path, and get that path relative to the save root.
I could use `Paths.get("foobar")` as well.

Reply:
In fact `--gamedir` would be just as "wrong" here, it would have to be the save path, which is not available at this point, as far as I can tell.